src/enclave/ServiceProvider/ServiceProvider.cpp

```
@@ -147,284 +151,197 @@ void ServiceProvider::export_public_key_code(const std::string &filename) {
+  int res = 0;
+  bool ret = false;
+  unsigned char* signer = NULL;
+
+
+  signer = (unsigned char*)malloc(signer_id_buf_size);
+  if (signer == NULL) {
+    printf("Out of memory\n");
+    goto exit;
   }
-}
 
-void ServiceProvider::connect_to_ias_helper(const std::string &ias_report_signing_ca_file) {
-  if (this->ias) {
-    return;
-  }
+  mbedtls_pk_init(&ctx);
 
-  std::unique_ptr<IAS_Connection> ias(
-    new IAS_Connection(is_production ? IAS_SERVER_PRODUCTION : IAS_SERVER_DEVELOPMENT, 0));
+  res = mbedtls_pk_parse_public_key(&ctx,
+                                    (const unsigned char*)signing_public_key_buf,
+                                    signing_public_key_buf_size+1);
 
-  const char *ias_client_cert_file = std::getenv("IAS_CLIENT_CERT_FILE");
-  if (!ias_client_cert_file) {
-    throw std::runtime_error(
-      "Set $IAS_CLIENT_CERT_FILE to the file generated by openssl x509, probably called "
-      "${OPAQUE_HOME}/client.crt.");
+  if (res != 0) {
+    printf("mbedtls_pk_parse_public_key failed with %d\n", res);
+    goto exit;
   }
-  ias->client_cert(ias_client_cert_file, "PEM");
-
-  const char *ias_client_cert_key_password_file = std::getenv("IAS_CLIENT_CERT_KEY_PASSWORD_FILE");
-  std::string password;
-  if (ias_client_cert_key_password_file) {
-    std::ifstream f(ias_client_cert_key_password_file);
-    std::stringstream s;
-    s << f.rdbuf();
-    password = s.str();
-  }
-  const char *ias_client_key_file = std::getenv("IAS_CLIENT_KEY_FILE");
-  if (!ias_client_key_file) {
-    throw std::runtime_error(
-      "Set $IAS_CLIENT_KEY_FILE to the file generated by openssl genrsa, probably called "
-      "${OPAQUE_HOME}/client.key.");
-  }
-  ias->client_key(ias_client_key_file, password.empty() ? nullptr : password.c_str());
 
-  ias->proxy_mode(IAS_PROXY_NONE);
-
-  ias->agent("wget");
+  pk_type = mbedtls_pk_get_type(&ctx);
+  if (pk_type != MBEDTLS_PK_RSA) {
+    printf("mbedtls_pk_get_type had incorrect type: %d\n", res);
```

Instead of printf can we instead throw a runtime error?